### PR TITLE
[go_router_builder] Generate initialLocation with StatefulShellBranchConfig

### PR DESCRIPTION
**Pull request by GeekAbdelouahed**
Add the possibility to generate `initialLocation` with `StatefulShellBranchConfig`

before:
```dart
class MyRoute extends StatefulShellBranchData {
  const MyRoute();
}
```

after:
```dart
class MyRoute extends StatefulShellBranchData {
  const MyRoute();

  static const String $initialLocation = '/my-route';
}
```

*List which issues are fixed by this PR. You must list at least one issue.*
[issues/130786](https://github.com/flutter/flutter/issues/130786#issuecomment-1712671560)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/go_router_builder/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 2.3.2
+* Add initialLocation with StatefulShellBranchConfig
```
Comment (hangyujin) [inline packages/go_router_builder/CHANGELOG.md:2]: nit: Adds

**Comment by hangyujin** [comment]
@GeekAbdelouahed  Thanks for the contribution! Can you also add tests to this PR?

**Comment by GeekAbdelouahed** [comment]
@hangyujin should I add a Widget test or Unit test?

**Review by hangyujin — APPROVED** [review]
LGTM, thanks!

File: packages/go_router_builder/test_inputs/statefull_shell_branch_data.dart.expect [hunk 2]
```diff
@@ -0,0 +1,3 @@
+RouteBase get $shellRouteBranchData => StatefulShellBranchData.$branch(
+      initialLocation: ShellRouteBranchData.$initialLocation,
+    );
```
Comment (chunhtai) [inline packages/go_router_builder/test_inputs/statefull_shell_branch_data.dart.expect:3]: new line

**Review by chunhtai — APPROVED** [review]
LGTM, just a format nit

**Pull request merged**